python/paddle/nn/functional/common.py

```diff
@@ -446,3 +447,195 @@ def _is_list_or_turple_(data):
+            pad = [1, 2, 1, 1]
+            input_data = np.arange(np.prod(input_shape), dtype=np.float32).reshape(input_shape) + 1
+            with dg.guard(place) as g:
+                input = dg.to_variable(input_data)
+                output = paddle.nn.functional.pad(input=input, pad=pad, mode="circular")
+                print(output.numpy())
+                # [[[[6. 4. 5. 6. 4. 5.]
+                #    [3. 1. 2. 3. 1. 2.]
+                #    [6. 4. 5. 6. 4. 5.]
+                #    [3. 1. 2. 3. 1. 2.]]]]
+    
+    """
+    data_format = data_format.upper()
+
+    input_dim = len(input.shape)
+
+    original_data_format = data_format
+    unsqueezed_dim = []
+
+    if isinstance(pad, Variable):
+        if data_format in ["NCL", "NCHW", "NCDHW"]:
+            data_format = "NCDHW"
+            if input_dim == 3:
+                pad = concat([zeros((4, ), dtype="int32"), pad], axis=0)
+                unsqueezed_dim = [3, 4]
+                input = unsqueeze(input, axes=unsqueezed_dim)
+            elif input_dim == 4:
+                pad = concat([pad, zeros((2, ), dtype="int32")], axis=0)
+                unsqueezed_dim = [2]
+                input = unsqueeze(input, axes=unsqueezed_dim)
+        elif data_format in ["NLC", "NHWC", "NDHWC"]:
+            data_format = "NDHWC"
+            if input_dim == 3:
+                pad = concat([zeros((4, ), dtype="int32"), pad], axis=0)
+                unsqueezed_dim = [2, 3]
+                input = unsqueeze(input, axes=unsqueezed_dim)
+            elif input_dim == 4:
+                pad = concat([pad, zeros((2, ), dtype="int32")], axis=0)
+                unsqueezed_dim = [1]
+                input = unsqueeze(input, axes=unsqueezed_dim)
+        else:
+            raise ValueError, "data_format should be in one of "
+            "[NCL, NCHW, NCDHW, NLC, NHWC, NDHWC] but got {}".format(
+                data_format)
+    else:
+        if data_format in ["NCL", "NCHW", "NCDHW"]:
+            data_format = "NCDHW"
+            if input_dim == 3:
+                pad = [0, 0, 0, 0] + pad
+                unsqueezed_dim = [3, 4]
+                input = unsqueeze(input, axes=unsqueezed_dim)
+            elif input_dim == 4:
+                pad = pad + [0, 0]
+                unsqueezed_dim = [2]
+                input = unsqueeze(input, axes=unsqueezed_dim)
+        elif data_format in ["NLC", "NHWC", "NDHWC"]:
+            data_format = "NDHWC"
+            if input_dim == 3:
+                pad = [0, 0, 0, 0] + pad
+                unsqueezed_dim = [2, 3]
+                input = unsqueeze(input, axes=unsqueezed_dim)
+            elif input_dim == 4:
+                pad = pad + [0, 0]
+                unsqueezed_dim = [1]
+                input = unsqueeze(input, axes=unsqueezed_dim)
+        else:
+            raise ValueError, "data_format should be in one of "
+            "[NCL, NCHW, NCDHW, NLC, NHWC, NDHWC] but got {}".format(
+                data_format)
+
```

Comment:
add fast path for dygraph mode. check the guide line for detailed info.

Reply:
done, thx